tests/framework/components/test_parser.py

```diff
@@ -75,40 +79,41 @@
 ]
 
 
-def mock_importer(path):
+def mock_importer(path: str) -> ABCMeta:
     return {
         "test_components.MockComponentA": MockComponentA,
         "test_components.MockComponentB": MockComponentB,
     }[path]
 
 
 @pytest.fixture()
-def parser(mocker) -> ComponentConfigurationParser:
+def parser(mocker: MockerFixture) -> ComponentConfigurationParser:
     parser = ComponentConfigurationParser()
-    parser.import_and_instantiate_component = mocker.Mock()
+    parser.import_and_instantiate_component = mocker.Mock()  # type: ignore [method-assign]
```

Comment:
obligatory "why exactly do you need to ignore this?"

Reply:
mypy doesn't like that we're assigning Mock to a method but since we're in pytest I don't think that restriction should apply